modules/jfx.incubator.input/src/main/java/com/sun/jfx/incubator/scene/control/input/EventHandlerPriority.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2023, 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.sun.jfx.incubator.scene.control.input;
+
+import java.util.Set;
+
+/**
+ * Codifies priority of event handler invocation.
+ */
+public enum EventHandlerPriority {
+    USER_HIGH(6000),
+    USER_KB(5000),
+    SKIN_KB(4000),
+    SKIN_HIGH(3000),
```

Comment:
Just curious about the logic behind the priorities. Like why for skin kb (is it keyboard?) if above the high priority but for user high is about kb? I understand that this is just a search order in which event handlers will be called and if higher priority handler consumes event it will not be proposed to the next priority level handler?

Reply:
excellent question!  

the idea is to have the handler for key bindings (KB) to be checked before other skin handlers, which typically deal with multiple keys or catch-all cases.